### DISCUSSION TITLE
Suggestion for the new authentication port number

### DISCUSSION THREAD

**Discussion by felipecrs** [blakeblackshear/frigate#11987] [Ideas]
Hi,

I think choosing `8080` is not a good number. `80` usually indicates http, of which is no longer true since TLS will be enabled by default on it. In this case, I think even `8443` is a better option, since it clearly indicates https.

Using clear numbers like this would help avoiding users make wrong assumptions about their protocols.

However, if I am understanding correctly, user can still choose whether the port will have TLS on or not. In this case, if user chooses TLS to be off, then 8443 will no longer make sense.

Therefore, I personally think any number beyond `8080` and `8443` would be better options, like for example `8000`.

What do you think? Has this been considered already? Thank you.

**Comment by blakeblackshear** [comment]
I hadn't thought about this. I guess it could be a good idea to choose a more random port number since 8080 is frequently used as well. Just used a random number generator to come up with `8971`. I don't see any conflicts with that port. Thoughts?

**Reply by felipecrs** [reply]
It looks good to me. This website, https://www.adminsub.net/tcp-udp-port-finder/8971, says it is already used by Apple Quicktime/iTunes for something and also another backup device solution.

However, it looks like any 8xxx port also says that. So, I think it's a good port. :)

**Reply by PovilasID** [reply]
Environmental variable that overrides the default port is a better option is the best way with dealing with this (not just relaying on docker port remapping). Also having a distinctive default port is handy then doing debugging and it will not get caught if you run frigate on in host mode or will avoid majority of other conflicts.

**Comment by leccelecce** [comment]
IANA designates 8080 (amongst several others) as "alternative" ports for HTTP - not HTTPS. So some applications or devices may explicitly expect HTTP traffic on port 8080. Ref: https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=http-alt

Apache Tomcat has used ports 8080/8443 as default for HTTP/HTTPS respectively for many years, serving on both by default. 

8080 for http/8443 for https would be a pretty safe option. Otherwise, picking a single port that is **not** 8080, 8180, 8443, 9443 etc...

**Comment by whatcheers** [comment]
Im running into this right now

**Reply by felipecrs** [reply]
Running into what?

**Reply by whatcheers** [reply]
Port conflict with 8080.  Doesn't seem to be anyway to change it....I had to change it in the docker.

**Reply by felipecrs** [reply]
Frigate doesn't use such port.

**Comment by LyraHarring** [comment]
You're right, [port 8080](https://cheapsslweb.com/blog/what-is-port-8080-http-port-80-vs-8080-vs-443-difference/) can be misleading since it's often assumed to be plain HTTP. Choosing a more neutral port like 8000 avoids protocol confusion, especially if TLS is optional. Using 8443 suggests HTTPS, which might mislead if TLS is disabled. Worth reconsidering for clarity. Hope it helps!
